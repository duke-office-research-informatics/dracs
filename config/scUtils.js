import React from "react";
import { ThemeProvider } from "styled-components";
import { mount, shallow } from "enzyme";
import { theme as defaultTheme } from "../lib/dracs.es.js";

export const shallowWithTheme = (children, customTheme) => {
  const theme = customTheme || defaultTheme;
  return shallow(children, { context: theme });
};

/* eslint-disable */
export function mountWithTheme(child) {
  return mount(child, {
    wrappingComponent: ({ children }) => (
      <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
    ),
  });
}

//FOR ACCESSING PORTALS VIA ENZYME -- import portal from dracs
// https://github.com/airbnb/enzyme/issues/1202#issuecomment-351639783
// function mountPortal(el){
//   const wrapper =  mountWithTheme(el);
//   wrapper.setState({ visible: true});
//   const portalInstance = wrapper.find(Portal).instance();
//   return mountWithTheme(portalInstance.props.children);
// }

//FOR ACCESSING EVENT LISTENERS ATTACHED TO DOCUMENT OR DOCUMENT.BODY
//https://github.com/airbnb/enzyme/issues/426#issuecomment-296304015
// const map = {};
// document.body.addEventListener = jest.fn((event, cb) => {
//   map[event] = cb;
// });
