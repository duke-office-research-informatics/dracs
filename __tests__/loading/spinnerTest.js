import React from "react";
import { mountWithTheme } from "../../config/scUtils.js";
import { axe } from "jest-axe";
import { Spinner, theme } from "../../lib/dracs.es.js";

describe("Loading Spinner", () => {
  it("matches snapshot and returns no a11y violations", async () => {
    const spinner = mountWithTheme(<Spinner />);
    expect(spinner).toMatchSnapshot();
    const results = await axe(spinner.html());
    expect(results).toHaveNoViolations();
  });
  it("sets the color of all 12 spinner bars via the color prop", () => {
    const testColor = theme.colors.error;
    const spinner = mountWithTheme(<Spinner color={testColor} />);
    expect(spinner).toMatchSnapshot();
    const spinnerBars = spinner.find("spinner__SpinnerBar");
    expect(spinnerBars.length).toBe(12);
    spinnerBars.forEach(bar => {
      expect(bar.props().color).toBe(testColor);
      expect(bar).toHaveStyleRule("background-color", testColor);
    });
  });
  it("sets an HTML id attr on the proper dom node via the id prop", () => {
    const testId = "testId";
    const spinner = mountWithTheme(<Spinner id={testId} />);
    expect(spinner).toMatchSnapshot();
    expect(spinner.find("spinner__SpinnerWrap").props().id).toBe(testId);
    expect(spinner.find("spinner__SpinnerWrap").is("#testId")).toBe(true);
  });
  it("sets the correct styles when margin prop is set to true", () => {
    const margin = "8px";
    const spinner = mountWithTheme(<Spinner margin={margin} />);
    expect(spinner).toMatchSnapshot();
    expect(spinner.find("spinner__ElementWrap")).toHaveStyleRule(
      "margin",
      margin
    );
  });
  it("sets the correct styles via size prop", () => {
    const spinner = mountWithTheme(<Spinner size={24} />);
    expect(spinner).toMatchSnapshot();
    expect(spinner.find("spinner__ElementWrap")).toHaveStyleRule(
      "height",
      "12px"
    );
    expect(spinner.find("spinner__ElementWrap")).toHaveStyleRule(
      "width",
      "12px"
    );
    expect(spinner.find("spinner__SpinnerWrap")).toHaveStyleRule(
      "height",
      "24px"
    );
    expect(spinner.find("spinner__SpinnerWrap")).toHaveStyleRule(
      "width",
      "24px"
    );
  });
  it("sets style on the proper dom node when style obj is passed via wrapperStyle prop", () => {
    const spinner = mountWithTheme(
      <Spinner wrapperStyle={{ padding: "8px" }} />
    );
    expect(spinner).toMatchSnapshot();
    expect(spinner.find("spinner__ElementWrap").props().style).toHaveProperty(
      "padding",
      "8px"
    );
  });
});
