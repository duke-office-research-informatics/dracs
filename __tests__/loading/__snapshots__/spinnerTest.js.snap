// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Spinner matches snapshot and returns no a11y violations 1`] = `
.c0 {
  position: relative;
  height: 16px;
  width: 16px;
  margin: 4px;
}

.c1 {
  position: absolute;
  top: 50%;
  left: 50%;
  width: 32px;
  height: 32px;
}

.c2 {
  -webkit-animation: jVfXJg 1.2s linear infinite;
  animation: jVfXJg 1.2s linear infinite;
  position: absolute;
  top: -3.9%;
  left: -10%;
  width: 20%;
  height: 7.8%;
  background-color: #0680cd;
  border-radius: 5px;
}

<div
  className="c0"
  size={32}
>
  <div
    className="c1"
    size={32}
  >
    <div
      className="c2"
      color="#0680cd"
      style={
        Object {
          "WebkitAnimationDelay": "-1.2s",
          "WebkitTransform": "rotate(0deg) translate(146%)",
          "animationDelay": "-1.2s",
          "transform": "rotate(0deg) translate(146%)",
        }
      }
    />
    <div
      className="c2"
      color="#0680cd"
      style={
        Object {
          "WebkitAnimationDelay": "-1.1s",
          "WebkitTransform": "rotate(30deg) translate(146%)",
          "animationDelay": "-1.1s",
          "transform": "rotate(30deg) translate(146%)",
        }
      }
    />
    <div
      className="c2"
      color="#0680cd"
      style={
        Object {
          "WebkitAnimationDelay": "-1s",
          "WebkitTransform": "rotate(60deg) translate(146%)",
          "animationDelay": "-1s",
          "transform": "rotate(60deg) translate(146%)",
        }
      }
    />
    <div
      className="c2"
      color="#0680cd"
      style={
        Object {
          "WebkitAnimationDelay": "-0.9s",
          "WebkitTransform": "rotate(90deg) translate(146%)",
          "animationDelay": "-0.9s",
          "transform": "rotate(90deg) translate(146%)",
        }
      }
    />
    <div
      className="c2"
      color="#0680cd"
      style={
        Object {
          "WebkitAnimationDelay": "-0.8s",
          "WebkitTransform": "rotate(120deg) translate(146%)",
          "animationDelay": "-0.8s",
          "transform": "rotate(120deg) translate(146%)",
        }
      }
    />
    <div
      className="c2"
      color="#0680cd"
      style={
        Object {
          "WebkitAnimationDelay": "-0.7s",
          "WebkitTransform": "rotate(150deg) translate(146%)",
          "animationDelay": "-0.7s",
          "transform": "rotate(150deg) translate(146%)",
        }
      }
    />
    <div
      className="c2"
      color="#0680cd"
      style={
        Object {
          "WebkitAnimationDelay": "-0.6s",
          "WebkitTransform": "rotate(180deg) translate(146%)",
          "animationDelay": "-0.6s",
          "transform": "rotate(180deg) translate(146%)",
        }
      }
    />
    <div
      className="c2"
      color="#0680cd"
      style={
        Object {
          "WebkitAnimationDelay": "-0.5s",
          "WebkitTransform": "rotate(210deg) translate(146%)",
          "animationDelay": "-0.5s",
          "transform": "rotate(210deg) translate(146%)",
        }
      }
    />
    <div
      className="c2"
      color="#0680cd"
      style={
        Object {
          "WebkitAnimationDelay": "-0.4s",
          "WebkitTransform": "rotate(240deg) translate(146%)",
          "animationDelay": "-0.4s",
          "transform": "rotate(240deg) translate(146%)",
        }
      }
    />
    <div
      className="c2"
      color="#0680cd"
      style={
        Object {
          "WebkitAnimationDelay": "-0.3s",
          "WebkitTransform": "rotate(270deg) translate(146%)",
          "animationDelay": "-0.3s",
          "transform": "rotate(270deg) translate(146%)",
        }
      }
    />
    <div
      className="c2"
      color="#0680cd"
      style={
        Object {
          "WebkitAnimationDelay": "-0.2s",
          "WebkitTransform": "rotate(300deg) translate(146%)",
          "animationDelay": "-0.2s",
          "transform": "rotate(300deg) translate(146%)",
        }
      }
    />
    <div
      className="c2"
      color="#0680cd"
      style={
        Object {
          "WebkitAnimationDelay": "-0.1s",
          "WebkitTransform": "rotate(330deg) translate(146%)",
          "animationDelay": "-0.1s",
          "transform": "rotate(330deg) translate(146%)",
        }
      }
    />
  </div>
</div>
`;

exports[`Loading Spinner sets an HTML id attr on the proper dom node via the id prop 1`] = `
.c0 {
  position: relative;
  height: 16px;
  width: 16px;
  margin: 4px;
}

.c1 {
  position: absolute;
  top: 50%;
  left: 50%;
  width: 32px;
  height: 32px;
}

.c2 {
  -webkit-animation: jVfXJg 1.2s linear infinite;
  animation: jVfXJg 1.2s linear infinite;
  position: absolute;
  top: -3.9%;
  left: -10%;
  width: 20%;
  height: 7.8%;
  background-color: #0680cd;
  border-radius: 5px;
}

<div
  className="c0"
  size={32}
>
  <div
    className="c1"
    id="testId"
    size={32}
  >
    <div
      className="c2"
      color="#0680cd"
      style={
        Object {
          "WebkitAnimationDelay": "-1.2s",
          "WebkitTransform": "rotate(0deg) translate(146%)",
          "animationDelay": "-1.2s",
          "transform": "rotate(0deg) translate(146%)",
        }
      }
    />
    <div
      className="c2"
      color="#0680cd"
      style={
        Object {
          "WebkitAnimationDelay": "-1.1s",
          "WebkitTransform": "rotate(30deg) translate(146%)",
          "animationDelay": "-1.1s",
          "transform": "rotate(30deg) translate(146%)",
        }
      }
    />
    <div
      className="c2"
      color="#0680cd"
      style={
        Object {
          "WebkitAnimationDelay": "-1s",
          "WebkitTransform": "rotate(60deg) translate(146%)",
          "animationDelay": "-1s",
          "transform": "rotate(60deg) translate(146%)",
        }
      }
    />
    <div
      className="c2"
      color="#0680cd"
      style={
        Object {
          "WebkitAnimationDelay": "-0.9s",
          "WebkitTransform": "rotate(90deg) translate(146%)",
          "animationDelay": "-0.9s",
          "transform": "rotate(90deg) translate(146%)",
        }
      }
    />
    <div
      className="c2"
      color="#0680cd"
      style={
        Object {
          "WebkitAnimationDelay": "-0.8s",
          "WebkitTransform": "rotate(120deg) translate(146%)",
          "animationDelay": "-0.8s",
          "transform": "rotate(120deg) translate(146%)",
        }
      }
    />
    <div
      className="c2"
      color="#0680cd"
      style={
        Object {
          "WebkitAnimationDelay": "-0.7s",
          "WebkitTransform": "rotate(150deg) translate(146%)",
          "animationDelay": "-0.7s",
          "transform": "rotate(150deg) translate(146%)",
        }
      }
    />
    <div
      className="c2"
      color="#0680cd"
      style={
        Object {
          "WebkitAnimationDelay": "-0.6s",
          "WebkitTransform": "rotate(180deg) translate(146%)",
          "animationDelay": "-0.6s",
          "transform": "rotate(180deg) translate(146%)",
        }
      }
    />
    <div
      className="c2"
      color="#0680cd"
      style={
        Object {
          "WebkitAnimationDelay": "-0.5s",
          "WebkitTransform": "rotate(210deg) translate(146%)",
          "animationDelay": "-0.5s",
          "transform": "rotate(210deg) translate(146%)",
        }
      }
    />
    <div
      className="c2"
      color="#0680cd"
      style={
        Object {
          "WebkitAnimationDelay": "-0.4s",
          "WebkitTransform": "rotate(240deg) translate(146%)",
          "animationDelay": "-0.4s",
          "transform": "rotate(240deg) translate(146%)",
        }
      }
    />
    <div
      className="c2"
      color="#0680cd"
      style={
        Object {
          "WebkitAnimationDelay": "-0.3s",
          "WebkitTransform": "rotate(270deg) translate(146%)",
          "animationDelay": "-0.3s",
          "transform": "rotate(270deg) translate(146%)",
        }
      }
    />
    <div
      className="c2"
      color="#0680cd"
      style={
        Object {
          "WebkitAnimationDelay": "-0.2s",
          "WebkitTransform": "rotate(300deg) translate(146%)",
          "animationDelay": "-0.2s",
          "transform": "rotate(300deg) translate(146%)",
        }
      }
    />
    <div
      className="c2"
      color="#0680cd"
      style={
        Object {
          "WebkitAnimationDelay": "-0.1s",
          "WebkitTransform": "rotate(330deg) translate(146%)",
          "animationDelay": "-0.1s",
          "transform": "rotate(330deg) translate(146%)",
        }
      }
    />
  </div>
</div>
`;

exports[`Loading Spinner sets style on the proper dom node when style obj is passed via wrapperStyle prop 1`] = `
.c0 {
  position: relative;
  height: 16px;
  width: 16px;
  margin: 4px;
}

.c1 {
  position: absolute;
  top: 50%;
  left: 50%;
  width: 32px;
  height: 32px;
}

.c2 {
  -webkit-animation: jVfXJg 1.2s linear infinite;
  animation: jVfXJg 1.2s linear infinite;
  position: absolute;
  top: -3.9%;
  left: -10%;
  width: 20%;
  height: 7.8%;
  background-color: #0680cd;
  border-radius: 5px;
}

<div
  className="c0"
  size={32}
  style={
    Object {
      "padding": "8px",
    }
  }
>
  <div
    className="c1"
    size={32}
  >
    <div
      className="c2"
      color="#0680cd"
      style={
        Object {
          "WebkitAnimationDelay": "-1.2s",
          "WebkitTransform": "rotate(0deg) translate(146%)",
          "animationDelay": "-1.2s",
          "transform": "rotate(0deg) translate(146%)",
        }
      }
    />
    <div
      className="c2"
      color="#0680cd"
      style={
        Object {
          "WebkitAnimationDelay": "-1.1s",
          "WebkitTransform": "rotate(30deg) translate(146%)",
          "animationDelay": "-1.1s",
          "transform": "rotate(30deg) translate(146%)",
        }
      }
    />
    <div
      className="c2"
      color="#0680cd"
      style={
        Object {
          "WebkitAnimationDelay": "-1s",
          "WebkitTransform": "rotate(60deg) translate(146%)",
          "animationDelay": "-1s",
          "transform": "rotate(60deg) translate(146%)",
        }
      }
    />
    <div
      className="c2"
      color="#0680cd"
      style={
        Object {
          "WebkitAnimationDelay": "-0.9s",
          "WebkitTransform": "rotate(90deg) translate(146%)",
          "animationDelay": "-0.9s",
          "transform": "rotate(90deg) translate(146%)",
        }
      }
    />
    <div
      className="c2"
      color="#0680cd"
      style={
        Object {
          "WebkitAnimationDelay": "-0.8s",
          "WebkitTransform": "rotate(120deg) translate(146%)",
          "animationDelay": "-0.8s",
          "transform": "rotate(120deg) translate(146%)",
        }
      }
    />
    <div
      className="c2"
      color="#0680cd"
      style={
        Object {
          "WebkitAnimationDelay": "-0.7s",
          "WebkitTransform": "rotate(150deg) translate(146%)",
          "animationDelay": "-0.7s",
          "transform": "rotate(150deg) translate(146%)",
        }
      }
    />
    <div
      className="c2"
      color="#0680cd"
      style={
        Object {
          "WebkitAnimationDelay": "-0.6s",
          "WebkitTransform": "rotate(180deg) translate(146%)",
          "animationDelay": "-0.6s",
          "transform": "rotate(180deg) translate(146%)",
        }
      }
    />
    <div
      className="c2"
      color="#0680cd"
      style={
        Object {
          "WebkitAnimationDelay": "-0.5s",
          "WebkitTransform": "rotate(210deg) translate(146%)",
          "animationDelay": "-0.5s",
          "transform": "rotate(210deg) translate(146%)",
        }
      }
    />
    <div
      className="c2"
      color="#0680cd"
      style={
        Object {
          "WebkitAnimationDelay": "-0.4s",
          "WebkitTransform": "rotate(240deg) translate(146%)",
          "animationDelay": "-0.4s",
          "transform": "rotate(240deg) translate(146%)",
        }
      }
    />
    <div
      className="c2"
      color="#0680cd"
      style={
        Object {
          "WebkitAnimationDelay": "-0.3s",
          "WebkitTransform": "rotate(270deg) translate(146%)",
          "animationDelay": "-0.3s",
          "transform": "rotate(270deg) translate(146%)",
        }
      }
    />
    <div
      className="c2"
      color="#0680cd"
      style={
        Object {
          "WebkitAnimationDelay": "-0.2s",
          "WebkitTransform": "rotate(300deg) translate(146%)",
          "animationDelay": "-0.2s",
          "transform": "rotate(300deg) translate(146%)",
        }
      }
    />
    <div
      className="c2"
      color="#0680cd"
      style={
        Object {
          "WebkitAnimationDelay": "-0.1s",
          "WebkitTransform": "rotate(330deg) translate(146%)",
          "animationDelay": "-0.1s",
          "transform": "rotate(330deg) translate(146%)",
        }
      }
    />
  </div>
</div>
`;

exports[`Loading Spinner sets the color of all 12 spinner bars via the color prop 1`] = `
.c0 {
  position: relative;
  height: 16px;
  width: 16px;
  margin: 4px;
}

.c1 {
  position: absolute;
  top: 50%;
  left: 50%;
  width: 32px;
  height: 32px;
}

.c2 {
  -webkit-animation: jVfXJg 1.2s linear infinite;
  animation: jVfXJg 1.2s linear infinite;
  position: absolute;
  top: -3.9%;
  left: -10%;
  width: 20%;
  height: 7.8%;
  background-color: #cc3300;
  border-radius: 5px;
}

<div
  className="c0"
  size={32}
>
  <div
    className="c1"
    size={32}
  >
    <div
      className="c2"
      color="#cc3300"
      style={
        Object {
          "WebkitAnimationDelay": "-1.2s",
          "WebkitTransform": "rotate(0deg) translate(146%)",
          "animationDelay": "-1.2s",
          "transform": "rotate(0deg) translate(146%)",
        }
      }
    />
    <div
      className="c2"
      color="#cc3300"
      style={
        Object {
          "WebkitAnimationDelay": "-1.1s",
          "WebkitTransform": "rotate(30deg) translate(146%)",
          "animationDelay": "-1.1s",
          "transform": "rotate(30deg) translate(146%)",
        }
      }
    />
    <div
      className="c2"
      color="#cc3300"
      style={
        Object {
          "WebkitAnimationDelay": "-1s",
          "WebkitTransform": "rotate(60deg) translate(146%)",
          "animationDelay": "-1s",
          "transform": "rotate(60deg) translate(146%)",
        }
      }
    />
    <div
      className="c2"
      color="#cc3300"
      style={
        Object {
          "WebkitAnimationDelay": "-0.9s",
          "WebkitTransform": "rotate(90deg) translate(146%)",
          "animationDelay": "-0.9s",
          "transform": "rotate(90deg) translate(146%)",
        }
      }
    />
    <div
      className="c2"
      color="#cc3300"
      style={
        Object {
          "WebkitAnimationDelay": "-0.8s",
          "WebkitTransform": "rotate(120deg) translate(146%)",
          "animationDelay": "-0.8s",
          "transform": "rotate(120deg) translate(146%)",
        }
      }
    />
    <div
      className="c2"
      color="#cc3300"
      style={
        Object {
          "WebkitAnimationDelay": "-0.7s",
          "WebkitTransform": "rotate(150deg) translate(146%)",
          "animationDelay": "-0.7s",
          "transform": "rotate(150deg) translate(146%)",
        }
      }
    />
    <div
      className="c2"
      color="#cc3300"
      style={
        Object {
          "WebkitAnimationDelay": "-0.6s",
          "WebkitTransform": "rotate(180deg) translate(146%)",
          "animationDelay": "-0.6s",
          "transform": "rotate(180deg) translate(146%)",
        }
      }
    />
    <div
      className="c2"
      color="#cc3300"
      style={
        Object {
          "WebkitAnimationDelay": "-0.5s",
          "WebkitTransform": "rotate(210deg) translate(146%)",
          "animationDelay": "-0.5s",
          "transform": "rotate(210deg) translate(146%)",
        }
      }
    />
    <div
      className="c2"
      color="#cc3300"
      style={
        Object {
          "WebkitAnimationDelay": "-0.4s",
          "WebkitTransform": "rotate(240deg) translate(146%)",
          "animationDelay": "-0.4s",
          "transform": "rotate(240deg) translate(146%)",
        }
      }
    />
    <div
      className="c2"
      color="#cc3300"
      style={
        Object {
          "WebkitAnimationDelay": "-0.3s",
          "WebkitTransform": "rotate(270deg) translate(146%)",
          "animationDelay": "-0.3s",
          "transform": "rotate(270deg) translate(146%)",
        }
      }
    />
    <div
      className="c2"
      color="#cc3300"
      style={
        Object {
          "WebkitAnimationDelay": "-0.2s",
          "WebkitTransform": "rotate(300deg) translate(146%)",
          "animationDelay": "-0.2s",
          "transform": "rotate(300deg) translate(146%)",
        }
      }
    />
    <div
      className="c2"
      color="#cc3300"
      style={
        Object {
          "WebkitAnimationDelay": "-0.1s",
          "WebkitTransform": "rotate(330deg) translate(146%)",
          "animationDelay": "-0.1s",
          "transform": "rotate(330deg) translate(146%)",
        }
      }
    />
  </div>
</div>
`;

exports[`Loading Spinner sets the correct styles via size prop 1`] = `
.c0 {
  position: relative;
  height: 12px;
  width: 12px;
  margin: 4px;
}

.c1 {
  position: absolute;
  top: 50%;
  left: 50%;
  width: 24px;
  height: 24px;
}

.c2 {
  -webkit-animation: jVfXJg 1.2s linear infinite;
  animation: jVfXJg 1.2s linear infinite;
  position: absolute;
  top: -3.9%;
  left: -10%;
  width: 20%;
  height: 7.8%;
  background-color: #0680cd;
  border-radius: 5px;
}

<div
  className="c0"
  size={24}
>
  <div
    className="c1"
    size={24}
  >
    <div
      className="c2"
      color="#0680cd"
      style={
        Object {
          "WebkitAnimationDelay": "-1.2s",
          "WebkitTransform": "rotate(0deg) translate(146%)",
          "animationDelay": "-1.2s",
          "transform": "rotate(0deg) translate(146%)",
        }
      }
    />
    <div
      className="c2"
      color="#0680cd"
      style={
        Object {
          "WebkitAnimationDelay": "-1.1s",
          "WebkitTransform": "rotate(30deg) translate(146%)",
          "animationDelay": "-1.1s",
          "transform": "rotate(30deg) translate(146%)",
        }
      }
    />
    <div
      className="c2"
      color="#0680cd"
      style={
        Object {
          "WebkitAnimationDelay": "-1s",
          "WebkitTransform": "rotate(60deg) translate(146%)",
          "animationDelay": "-1s",
          "transform": "rotate(60deg) translate(146%)",
        }
      }
    />
    <div
      className="c2"
      color="#0680cd"
      style={
        Object {
          "WebkitAnimationDelay": "-0.9s",
          "WebkitTransform": "rotate(90deg) translate(146%)",
          "animationDelay": "-0.9s",
          "transform": "rotate(90deg) translate(146%)",
        }
      }
    />
    <div
      className="c2"
      color="#0680cd"
      style={
        Object {
          "WebkitAnimationDelay": "-0.8s",
          "WebkitTransform": "rotate(120deg) translate(146%)",
          "animationDelay": "-0.8s",
          "transform": "rotate(120deg) translate(146%)",
        }
      }
    />
    <div
      className="c2"
      color="#0680cd"
      style={
        Object {
          "WebkitAnimationDelay": "-0.7s",
          "WebkitTransform": "rotate(150deg) translate(146%)",
          "animationDelay": "-0.7s",
          "transform": "rotate(150deg) translate(146%)",
        }
      }
    />
    <div
      className="c2"
      color="#0680cd"
      style={
        Object {
          "WebkitAnimationDelay": "-0.6s",
          "WebkitTransform": "rotate(180deg) translate(146%)",
          "animationDelay": "-0.6s",
          "transform": "rotate(180deg) translate(146%)",
        }
      }
    />
    <div
      className="c2"
      color="#0680cd"
      style={
        Object {
          "WebkitAnimationDelay": "-0.5s",
          "WebkitTransform": "rotate(210deg) translate(146%)",
          "animationDelay": "-0.5s",
          "transform": "rotate(210deg) translate(146%)",
        }
      }
    />
    <div
      className="c2"
      color="#0680cd"
      style={
        Object {
          "WebkitAnimationDelay": "-0.4s",
          "WebkitTransform": "rotate(240deg) translate(146%)",
          "animationDelay": "-0.4s",
          "transform": "rotate(240deg) translate(146%)",
        }
      }
    />
    <div
      className="c2"
      color="#0680cd"
      style={
        Object {
          "WebkitAnimationDelay": "-0.3s",
          "WebkitTransform": "rotate(270deg) translate(146%)",
          "animationDelay": "-0.3s",
          "transform": "rotate(270deg) translate(146%)",
        }
      }
    />
    <div
      className="c2"
      color="#0680cd"
      style={
        Object {
          "WebkitAnimationDelay": "-0.2s",
          "WebkitTransform": "rotate(300deg) translate(146%)",
          "animationDelay": "-0.2s",
          "transform": "rotate(300deg) translate(146%)",
        }
      }
    />
    <div
      className="c2"
      color="#0680cd"
      style={
        Object {
          "WebkitAnimationDelay": "-0.1s",
          "WebkitTransform": "rotate(330deg) translate(146%)",
          "animationDelay": "-0.1s",
          "transform": "rotate(330deg) translate(146%)",
        }
      }
    />
  </div>
</div>
`;

exports[`Loading Spinner sets the correct styles when margin prop is set to true 1`] = `
.c0 {
  position: relative;
  height: 16px;
  width: 16px;
  margin: 8px;
}

.c1 {
  position: absolute;
  top: 50%;
  left: 50%;
  width: 32px;
  height: 32px;
}

.c2 {
  -webkit-animation: jVfXJg 1.2s linear infinite;
  animation: jVfXJg 1.2s linear infinite;
  position: absolute;
  top: -3.9%;
  left: -10%;
  width: 20%;
  height: 7.8%;
  background-color: #0680cd;
  border-radius: 5px;
}

<div
  className="c0"
  size={32}
>
  <div
    className="c1"
    size={32}
  >
    <div
      className="c2"
      color="#0680cd"
      style={
        Object {
          "WebkitAnimationDelay": "-1.2s",
          "WebkitTransform": "rotate(0deg) translate(146%)",
          "animationDelay": "-1.2s",
          "transform": "rotate(0deg) translate(146%)",
        }
      }
    />
    <div
      className="c2"
      color="#0680cd"
      style={
        Object {
          "WebkitAnimationDelay": "-1.1s",
          "WebkitTransform": "rotate(30deg) translate(146%)",
          "animationDelay": "-1.1s",
          "transform": "rotate(30deg) translate(146%)",
        }
      }
    />
    <div
      className="c2"
      color="#0680cd"
      style={
        Object {
          "WebkitAnimationDelay": "-1s",
          "WebkitTransform": "rotate(60deg) translate(146%)",
          "animationDelay": "-1s",
          "transform": "rotate(60deg) translate(146%)",
        }
      }
    />
    <div
      className="c2"
      color="#0680cd"
      style={
        Object {
          "WebkitAnimationDelay": "-0.9s",
          "WebkitTransform": "rotate(90deg) translate(146%)",
          "animationDelay": "-0.9s",
          "transform": "rotate(90deg) translate(146%)",
        }
      }
    />
    <div
      className="c2"
      color="#0680cd"
      style={
        Object {
          "WebkitAnimationDelay": "-0.8s",
          "WebkitTransform": "rotate(120deg) translate(146%)",
          "animationDelay": "-0.8s",
          "transform": "rotate(120deg) translate(146%)",
        }
      }
    />
    <div
      className="c2"
      color="#0680cd"
      style={
        Object {
          "WebkitAnimationDelay": "-0.7s",
          "WebkitTransform": "rotate(150deg) translate(146%)",
          "animationDelay": "-0.7s",
          "transform": "rotate(150deg) translate(146%)",
        }
      }
    />
    <div
      className="c2"
      color="#0680cd"
      style={
        Object {
          "WebkitAnimationDelay": "-0.6s",
          "WebkitTransform": "rotate(180deg) translate(146%)",
          "animationDelay": "-0.6s",
          "transform": "rotate(180deg) translate(146%)",
        }
      }
    />
    <div
      className="c2"
      color="#0680cd"
      style={
        Object {
          "WebkitAnimationDelay": "-0.5s",
          "WebkitTransform": "rotate(210deg) translate(146%)",
          "animationDelay": "-0.5s",
          "transform": "rotate(210deg) translate(146%)",
        }
      }
    />
    <div
      className="c2"
      color="#0680cd"
      style={
        Object {
          "WebkitAnimationDelay": "-0.4s",
          "WebkitTransform": "rotate(240deg) translate(146%)",
          "animationDelay": "-0.4s",
          "transform": "rotate(240deg) translate(146%)",
        }
      }
    />
    <div
      className="c2"
      color="#0680cd"
      style={
        Object {
          "WebkitAnimationDelay": "-0.3s",
          "WebkitTransform": "rotate(270deg) translate(146%)",
          "animationDelay": "-0.3s",
          "transform": "rotate(270deg) translate(146%)",
        }
      }
    />
    <div
      className="c2"
      color="#0680cd"
      style={
        Object {
          "WebkitAnimationDelay": "-0.2s",
          "WebkitTransform": "rotate(300deg) translate(146%)",
          "animationDelay": "-0.2s",
          "transform": "rotate(300deg) translate(146%)",
        }
      }
    />
    <div
      className="c2"
      color="#0680cd"
      style={
        Object {
          "WebkitAnimationDelay": "-0.1s",
          "WebkitTransform": "rotate(330deg) translate(146%)",
          "animationDelay": "-0.1s",
          "transform": "rotate(330deg) translate(146%)",
        }
      }
    />
  </div>
</div>
`;
